portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -82,36 +84,65 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery commentHistory = datastore.prepare(commentHistoryQuery);
     List<Entity> commentHistoryList = commentHistory.asList(FetchOptions.Builder.withDefaults());
     int totalComments = commentHistoryList.size();
-    int totalPages;
 
     // Obtain the maximum number of comments to display. If the number is invalid, take the default value.
+    String maxCommentsToDisplayStr = request.getParameter("maxCommentsToDisplay");
     int maxCommentsToDisplay;
     try {
-      maxCommentsToDisplay = Integer.parseInt(request.getParameter("maxCommentsToDisplay"));
+      maxCommentsToDisplay = Integer.parseInt(maxCommentsToDisplayStr);
       if (maxCommentsToDisplay < 0) {
```

Comment:
Should this be < 1?

Reply:
The original behavior is that users can choose to not display any comments. Do you think I should remove the permission for maxCommentsToDisplay=0?

Reply:
Ah gotcha.  I wonder if it'd be better to just include a way to toggle the whole comment section on/off instead?  (if that's too hard, feel free to leave as is)

Reply:
Got it! I think that's doable. I'll try it. Then I'll remove the permission for 0 comments and later add a on/off checkbox or something

Reply:
I added the on/off checkbox feature. It behaves as the following:
- When users check the checkbox to disable the comment history section, the entire comment history section disappears while everything that was displayed gets backed up (eg. the 2nd page, with 3 comments).
- When users uncheck the checkbox, the comment history section is restored to what it looked like before. Users can adjust the number of comments to show per page and the page ID, in the same way as before.